client/additional-methods-setup/upe-preview-methods-selector/add-payment-methods-task.js

```diff
@@ -0,0 +1,258 @@
+/**
+ * External dependencies
+ */
+import React, {
+	useCallback,
+	useContext,
+	useEffect,
+	useMemo,
+	useState,
+} from 'react';
+import { __ } from '@wordpress/i18n';
+import { Button, Card, CardBody } from '@wordpress/components';
+import interpolateComponents from 'interpolate-components';
+
+/**
+ * Internal dependencies
+ */
+import WizardTaskContext from '../wizard/task/context';
+import CollapsibleBody from '../wizard/collapsible-body';
+import WizardTaskItem from '../wizard/task-item';
+import {
+	useEnabledPaymentMethodIds,
+	useGetAvailablePaymentMethodIds,
+	useSettings,
+} from '../../data';
+import PaymentMethodCheckboxes from '../../components/payment-methods-checkboxes';
+import PaymentMethodCheckbox from '../../components/payment-methods-checkboxes/payment-method-checkbox';
+import { LoadableBlock } from '../../components/loadable';
+import LoadableSettingsSection from '../../settings/loadable-settings-section';
+import CurrencyInformationForMethods from '../../components/currency-information-for-methods';
+
+const usePaymentMethodsCheckboxState = () => {
+	const availablePaymentMethods = useGetAvailablePaymentMethodIds();
+	const [ paymentMethodsState, setPaymentMethodsState ] = useState( {} );
+
+	useEffect( () => {
+		setPaymentMethodsState(
+			// by default, all the checkboxes should be "checked"
+			availablePaymentMethods
+				.filter( ( method ) =>
+					[ 'giropay', 'sofort', 'sepa_debit' ].includes( method )
+				)
+				.reduce(
+					( map, paymentMethod ) => ( {
+						...map,
+						[ paymentMethod ]: true,
+					} ),
+					{}
+				)
+		);
+	}, [ availablePaymentMethods, setPaymentMethodsState ] );
+
+	const handleChange = useCallback(
+		( paymentMethodName, enabled ) => {
+			setPaymentMethodsState( ( oldValues ) => ( {
+				...oldValues,
+				[ paymentMethodName ]: enabled,
+			} ) );
+		},
+		[ setPaymentMethodsState ]
+	);
+
+	return [ paymentMethodsState, handleChange ];
+};
+
+const ContinueButton = ( { paymentMethodsState } ) => {
+	const { setCompleted } = useContext( WizardTaskContext );
+	const [
+		initialEnabledPaymentMethodIds,
+		updateEnabledPaymentMethodIds,
+	] = useEnabledPaymentMethodIds();
+
+	const { saveSettings, isSaving } = useSettings();
+
+	const checkedPaymentMethods = useMemo(
+		() =>
+			Object.entries( paymentMethodsState )
+				.map( ( [ method, enabled ] ) => enabled && method )
+				.filter( Boolean ),
+		[ paymentMethodsState ]
+	);
+
+	const unCheckedPaymentMethods = Object.entries( paymentMethodsState )
+		.map( ( [ method, enabled ] ) => ! enabled && method )
+		.filter( Boolean );
```

Comment:
I am curious what the `&& method` do here? I tried removing it and it seems to give me the same output. Also, is it a best practice to always run `.filter(Boolean)` on true/flag flags? I do noticed `[1,2,3,'asd', true, '', 'true'].filter(Boolean)` returns `[1, 2, 3, "asd", true, "true"]`, is the purpose of this to remove blank strings? Is this serving more purposes? 

Genuinely interested.

Reply:
If the method is enabled, it returns its name, otherwise it returns `false`. Then all the disabled methods are removed from the array.

It's basically a shortcut to:
```
if(enabled) return method;
return false
```